### ISSUE TITLE
zoneinfo.gettz() objects are singletons for some reason

### ISSUE THREAD

**Issue by pganssle** [dateutil/dateutil#226]
#When exploring why the tests were failing on #225, I discovered that `tzfile()` objects constructed through `zoneinfo.gettz()` are completely immune to being copied - not entirely sure why:

```python
import copy
import dateutil
from dateutil import tz
from dateutil.zoneinfo import gettz

zone = 'Australia/Canberra'
z_t = tz.gettz(zone)
z_z = gettz(zone)

t0_t = z_t is z_t                           # True
t0_z = z_z is z_z                           # True

t1_t = z_t is tz.gettz(zone)                # False
t1_z = z_z is gettz(zone)                   # True (fair enough)

t2_t = z_t is copy.copy(z_t)                # False
t2_z = z_z is copy.copy(z_z)                # True (Curious...)

t3_t = z_t is copy.deepcopy(z_t)            # False
t3_z = z_z is copy.deepcopy(z_z)            # True (WTF?)

print("Dateutil version: {}".format(dateutil.__version__))     # 2.5.1
for tt, tz, st in [(t0_t, t0_z, 'same object: {}'),
                   (t1_t, t1_z, 'new construction: {}'),
                   (t2_t, t2_z, 'shallow copy: {}'),
                   (t3_t, t3_z, 'deep copy: {}')]:
    print("tz.gettz(), " + st.format(tt))
    print("zoneinfo.gettz(), " + st.format(tz))
```

This will probably need to be addressed as part of #225.

**Comment by pganssle** [comment]
I have a partial fix in #225. I don't totally know why it was happening - something to do with how the file was opened, I believe - but I added explicit `__copy__` and `__deepcopy__` functions to fix it. However, I think that it's probably a good idea to switch to a `__getstate__`/`__setstate__` based solution, as this will also broaden the scope of what types of `tzfile` objects can be pickled.

**Comment by pganssle** [comment]
Fixed in #225.
